apps/gdalmdiminfo_lib.cpp

```diff
@@ -1060,6 +1061,113 @@ static void WriteToStdout(const char *pszText, void *)
+{
+    auto argParser = std::make_unique<GDALArgumentParser>(
+        "gdalmdiminfo", /* bForBinary=*/psOptionsForBinary != nullptr);
+
+    argParser->add_description(
+        _("Lists various information about a GDAL multidimensional dataset."));
+
+    argParser->add_epilog(_("For more details, consult "
+                            "https://gdal.org/programs/gdalmdiminfo.html"));
+
+    argParser->add_argument("-detailed")
+        .flag()
+        .store_into(psOptions->bDetailed)
+        .help(_("Most verbose output. Report attribute data types and array "
+                "values."));
+
+    argParser->add_inverted_logic_flag(
+        "-nopretty", &psOptions->bPretty,
+        _("Outputs on a single line without any indentation."));
+
+    argParser->add_argument("-array")
+        .metavar("<array_name>")
+        .store_into(psOptions->osArrayName)
+        .help(_("Name of the array, used to restrict the output to the "
+                "specified array."));
+
+    argParser->add_argument("-limit")
+        .metavar("<number>")
+        .scan<'i', int>()
+        .store_into(psOptions->nLimitValuesByDim)
+        .help(_("Number of values in each dimension that is used to limit the "
+                "display of array values."));
+
+    if (psOptionsForBinary)
+    {
+        argParser->add_open_options_argument(
+            psOptionsForBinary->aosOpenOptions);
+
+        argParser->add_argument("-if")
+            .append()
+            .metavar("<format>")
+            .action(
+                [psOptionsForBinary](const std::string &s)
+                {
+                    if (GDALGetDriverByName(s.c_str()) == nullptr)
+                    {
+                        CPLError(CE_Warning, CPLE_AppDefined,
+                                 "%s is not a recognized driver", s.c_str());
+                    }
+                    psOptionsForBinary->aosAllowInputDrivers.AddString(
+                        s.c_str());
+                })
+            .help(
+                _("Format/driver name(s) to try when opening the input file."));
+
+        argParser->add_argument("dataset_name")
+            .metavar("<dataset_name>")
+            .store_into(psOptionsForBinary->osFilename)
+            .help("Input dataset.");
+    }
+
+    argParser->add_argument("-arrayoption")
+        .metavar("<NAME=VALUE>")
+        .append()
+        .action([psOptions](const std::string &s)
+                { psOptions->aosArrayOptions.AddString(s.c_str()); })
+        .help(_("Option passed to GDALGroup::GetMDArrayNames() to filter "
+                "reported arrays."));
+
+    argParser->add_argument("-stats")
+        .flag()
+        .store_into(psOptions->bStats)
+        .help(_("Read and display image statistics."));
+
+    argParser->add_argument("-stdout")
+        .flag()
+        .store_into(psOptions->bStdoutOutput)
+        .help(_("Write output to stdout instead of a JSON file."));
```

Comment:
```suggestion
    // Only used by gdalmdiminfo binary to write output to stdout instead of in a string, in JSON mode
    argParser->add_argument("-stdout")
        .flag()
        .hidden()
        .store_into(psOptions->bStdoutOutput);
```

Reply:
done